src/com/TMS/Lesson9/ServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.TMS.Lesson9;
+
+import java.util.*;
+
+public class ServiceImpl implements Service {
+
+    @Override
+    public List<User> findByName(List<User> list, String name) {
+        List<User> finalList = new ArrayList<>();
+        for (User u : list) {
+            if (u.getName().equals(name)) {
+                finalList.add(u);
+            }
+        }
+        return finalList;
+    }
+
+    @Override
+    public Set<User> findByGender(Set<User> set, Gender gender) {
+        Set<User> finalSet = new TreeSet<>();
+        for (User u : set) {
+            if (u.getGender().name().equals(gender.name())) {
+                finalSet.add(u);
+            }
+        }
+        return finalSet;
+    }
+
+    @Override
+    public List<User> sortByAge(List<User> list) {
+        boolean flag = true;
```

Comment:
instead of using you approach just convert list to treeset and put correct comparator